src/structures/game_session.ts

```diff
@@ -142,6 +142,12 @@ export default class GameSession {
     /** Mapping of user ID to bookmarked songs, uses Map since Set doesn't remove QueriedSong duplicates */
     private bookmarkedSongs: { [userID: string]: Map<string, QueriedSong> };
 
+    /** List of premium participants in the GameSession */
+    private premiumParticipants: Set<string>;
```

Comment:
Instead of managing this state, could we just compare the current members in voice channel with our premium_users table? Have some method called `updatePremiumStatus` which is called on every voice channel leave/switch/join, and it just compares the current members and premium_users, and updates `this.premiumGame` accordingly.

Reply:
This will be a bit more expensive but I will make the change since it simplifies it so much